packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -0,0 +1,497 @@
+								this.id,
+								'.signatures',
+							),
+						];
+					}
+				}
+
+				return errorArray;
+			},
+			[],
+		);
+
+		return {
+			id: this.id,
+			status: validateErrors.length === 0 ? Status.OK : Status.FAIL,
+			errors: validateErrors,
+		};
+	}
+
+	public getRequiredAttributes(): object {
+		return {
+			ACCOUNTS: [cryptography.getAddressFromPublicKey(this.senderPublicKey)],
+		};
+	}
+
+	public verify(sender: Account): TransactionResponse {
+		// Balance verification
+		const { errors: balanceError } = checkBalance(sender, this.fee);
+
+		// Check transaction against account
+		const transactionErrors = Object.entries(sender).reduce(
+			(
+				errorArray: ReadonlyArray<TransactionError>,
+				property: ReadonlyArray<string | number>,
+			): ReadonlyArray<TransactionError> => {
+				const [key, value] = property;
+
+				// Check senderPublicKey
+				if (key === 'publicKey' && value !== this.senderPublicKey) {
+					return [
+						...errorArray,
+						new TransactionError(
+							'Invalid sender publicKey',
+							this.id,
+							'.senderPublicKey',
+						),
+					];
+				}
+
+				// Check for missing signSignature
+				if (key === 'secondPublicKey' && value && !this.signSignature) {
+					return [
+						...errorArray,
+						new TransactionError(
+							'Missing signSignature',
+							this.id,
+							'.signSignature',
+						),
+					];
+				}
+
+				// Check signatures on transaction
+				if (
+					key === 'multisignatures' &&
+					Array.isArray(value) &&
+					value.length > 0 &&
+					(!this.signatures ||
+						(this.signatures && this.signatures.length === 0))
+				) {
+					return [
+						...errorArray,
+						new TransactionError('Missing signatures', this.id, '.signatures'),
+					];
+				}
+
+				// Check if transaction has enough signatures to be confirmed
+				if (
+					key === 'multimin' &&
+					this.signatures &&
+					this.signatures.length < value
+				) {
+					return [
+						...errorArray,
+						new TransactionError('Missing signatures', this.id, '.signatures'),
+					];
+				}
+
+				// Check senderId
+				if (
+					key === 'address' &&
+					typeof value === 'string' &&
+					value.toUpperCase() !== this.senderId.toUpperCase()
+				) {
+					return [
+						...errorArray,
+						new TransactionError(
+							'Invalid sender address',
+							this.id,
+							'.senderId',
+						),
+					];
+				}
+
+				return errorArray;
+			},
+			[],
+		);
+
+		// Check missing secondPublicKey on account
+		const missingSecondPublicKeyError =
+			this.signSignature && !sender.secondPublicKey
+				? [
+						new TransactionError(
+							'Sender does not have a secondPublicKey',
+							this.id,
+						),
+				  ]
+				: [];
+
+		// Signature verifications
+
+		const transactionHash = cryptography.hash(this.getBasicBytes());
+
+		// Verify signSignature
+		const signSignatureError =
+			sender.secondPublicKey &&
+			!cryptography.verifyData(
+				transactionHash,
```

Comment:
To check second signature, I think first signature needs to be included in the hash. Please check the core